docs: add 2023-04-03 meeting notes (#69)
nodejs/performance: meetings/2023-04-03.md

# Node.js  Performance Team Meeting 2023-04-03

## Links

* **Recording**:  https://www.youtube.com/watch?v=MVqHVCv-rnQ
* **GitHub Issue**: https://github.com/nodejs/performance/issues/68

## Present

* Performance team: @nodejs/performance
* Yagiz Nizipli: @anonrig
* Stephen Belanger: @qard
* Robert Nagy: @ronag
* Debadree Chatterjee @debadree25

## Agenda

## Announcements

* Ada v2 is released. Announcement post is available at http://yagiz.co/announcing-ada-url-parser-v2-0/

> * V8 11.3 brings `String.prototype.toWellFormed()` support with 70-80% faster `toUSVString`

Reviewer: Do we have a benchmark for this?

Author: Yes: https://ci.nodejs.org/view/Node.js%20benchmark/job/benchmark-node-micro-benchmarks/1315/

> * Increasing the default watermark size for version 20. It provides a performance boost! 

Reviewer: ```suggestion
* Increasing the default watermark size for version 20. It provides a performance boost!
```

*Extracted from **performance-agenda** labelled issues and pull requests from the **nodejs org** prior to the meeting.

### nodejs/performance

* add/remove handlers are slow on EventTarget [#60](https://github.com/nodejs/performance/issues/60)

>   * It would be great to add a fast path for AbortSignal to have only one AbortController.
>   * There are no other works on it right now.
>   * Debadree attempted on adding a hashmap based approach but did not help with any performance impacts.
>   * It would be great to have a benchmark for adding and removing a single event listener (the current benchmarks adds and removes multiple benchmarks), because the proposed benchmark would never traverse the linkedlist.

Reviewer 2: ```suggestion
  * It would be great to add a fast path for AbortSignal to have only one `'abort`' listener.
  * There is no one working on it right now.
  * Debadree attempted on adding a hashmap based approach but did not help with any performance impacts.
  * It would be great to have a benchmark for adding and removing a single event listener (the current benchmarks adds multiple listeners and then dispatches), because the proposed benchmark would never traverse the linkedlist.
```

* Session proposals for the next Collaborator Summit - what should we submit [#64](https://github.com/nodejs/performance/issues/64)
* Bun & Deno Comparison [#55](https://github.com/nodejs/performance/issues/55)
  * Yagiz proposed adding his own performance benchmarks to the performance repository. Will keep this for visibility.
* AbortSignal pattern is slow [#44](https://github.com/nodejs/performance/issues/44)
  * Closing it since it is a duplicate of add/remove handlers for EventTarget
* Module resolution [#39](https://github.com/nodejs/performance/issues/39)
  * Put this on hold until off-thread loader pull request is landed.
* EventTarget [#32](https://github.com/nodejs/performance/issues/32)
  * Closing it due to Santiago’s amazing contributions.
* fetch [#11](https://github.com/nodejs/performance/issues/11)

>   * Robert: I don’t know where to start optimizing. There’s more work to be done. Maybe we can ask Rafael on making a new analysis on undici fetch?

Author: @RafaelGSS fyi
